src/System.Diagnostics.Process/tests/ProcessTests.Unix.cs

```diff
@@ -816,6 +854,36 @@ private void RunTestAsSudo(Func<string, int> testMethod, string arg)
         [DllImport("libc")]
         private static extern uint geteuid();
 
+        [DllImport("libc")]
+        private static extern uint getuid();
+
+        [DllImport("libc")]
+        private static extern uint getgid();
+
+        [DllImport("libc", SetLastError = true)]
+        private unsafe static extern int getgroups(int size, uint* list);
+
+        private unsafe static uint[] GetGroups()
+        {
+            int maxSize = 128;
+            Span<uint> groups = stackalloc uint[maxSize];
+            fixed (uint* pGroups = groups)
+            {
+                int rv = getgroups(maxSize, pGroups);
+                if (rv == -1)
+                {
+                    // If this throws with EINVAL, maxSize should be increased.
```

Comment:
NGROUPS_MAX seems to be up to 65536. Should we retry with a larger buffer if we get EINVAL?

Reply:
Oh, it's just the tests.

Reply:
Yes 128 should be a safe value and I didn't want to make this code more complicated.